sympy/matrices/sparse.py

```diff
@@ -985,8 +985,10 @@ def col_join(self, other):
         >>> C == A.row_insert(A.rows, Matrix(B))
         True
         """
-        if not self:
-            return type(self)(other)
+        # A null matrix can always be stacked (see  #10770)
+        if self.rows == 0 and self.cols != other.cols:
```

Comment:
It looks like the result will be a copy of `other` in either case. Would it be possible to return that here?